lessons/pt/testing/stream_data.md

```diff
@@ -0,0 +1,466 @@
+%{
+  version: "1.1.1",
+  title: "StreamData",
+  excerpt: """
+  Uma biblioteca de testes unitários baseada em exemplos como a [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) é uma ferramenta maravilhosa para verificar se o código que você escreveu funciona da maneira que você espera.
+  Entretanto, esse tipo de teste tem algumas desvantagens:
+
+  * Pode ser fácil de perder casos de borda, já que você está testando apenas algumas poucas entradas.
+  * Você pode escrever esses testes sem pensar cuidadosamente em seus requisitos.
+  * Esses testes podem ser bastante verbosos quando você usa vários exemplos para uma função.
+
+  Nessa lição, vamos explorar como [StreamData](https://github.com/whatyouhide/stream_data) pode nos ajudar a superar alguns desses problemas.
+  """
+}
+---
+
+## O que é StreamData?
+
+[StreamData](https://github.com/whatyouhide/stream_data) é uma biblioteca que realiza testes sem estado (stateless), baseados em propriedade.
+
+A biblioteca StreamData vai rodar cada teste [100 vezes por padrão](https://hexdocs.pm/stream_data/ExUnitProperties.html#check/1-options), usando dados aleatórios.
+Quando um teste falha, a biblioteca vai tentar reduzir [(shrink)](https://hexdocs.pm/stream_data/ExUnitProperties.html#check/1-options) a entrada ao menor valor que causa a falha do teste.
+Isso pode ser bastante útil na hora de debuggar seu código!
+Se, por exemplo, uma lista de 50 elementos faz a sua função quebrar e apenas um elemento dessa lista é problemático, StreamData pode te ajudar a identificar o elemento que causa o problema.
+
+Essa biblioteca de testes tem dois módulos principais.
+[`StreamData`](https://hexdocs.pm/stream_data/StreamData.html) gera streams de dados aleatórios.
+[`ExUnitProperties`](https://hexdocs.pm/stream_data/ExUnitProperties.html) permite que você rode seus testes em suas funções, usando os dados gerados como dados de entrada.
+
+Você pode estar se perguntando como podem ser gerados dados que façam sentido para a função que você está testando, se não se sabe exatamente qual é o input esperado. Continue lendo!
```

Comment:
I am not 100% sure if it is the best translation for this sentence. I would appreciate if you guys could double check it :)
The english version is:
> You might be asking how you can say anything meaningful about a function if you don't know what your exact inputs are. Read on!

Reply:
It looks fine to me